### PR TITLE
add zulu jdk plugin

### PR DESCRIPTION
**Pull request by yimiaoxiehou**
support install jdk via [zulu ](https://www.azul.com/downloads/#zulu)

just test passwd in ubuntu22(amd64). have not other test env.

test output:
```
root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox  add  --source ./azul-jdk.lua
Loading plugin from ./azul-jdk.lua...
Checking plugin...
Plugin info:
Name    -> java
Author  -> yimiaoxiehou
Version -> 0.0.1
Desc    -> Azul JDK, also known as Zulu
Path    -> /root/.version-fox/plugin/java.lua
Add java plugin successfully! 
Please use `vfox install java@<version>` to install the version you need.
root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox ls
All installed sdk versions
├──java
├─┬nodejs
│ └──v21.6.1
└─┬golang
  └──v1.21.6

root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox search
sdk name is required
root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox search java
Please select a version of java [type to search]: 
   v21 (LTS)
   v20
   v19
   v18
   v17 (LTS)
   v13
->  v11 (LTS)
   v8 (LTS)
   v7 (LTS)
   v6 (LTS)
Press ↑/↓ to select and press ←/→ to page, and press Enter to confirm
Installing java@11...
Downloading... 100% [==========================================================================================================================================] (2.3 MB/s)           
Verifying checksum bd0f4e71328854d3dc21c98c88cccaabcc64353e2d6bcd6da89198a303d246b7...
Unpacking /root/.version-fox/cache/java/zulu11.70.15-ca-jdk11.0.22-linux_x64.zip...
Install java@11 success! 
Please use vfox use java@11 to use it.
root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox use java@11
Warning: The current shell lacks hook support or configuration. It has switched to global scope automatically.
Now using java@11.
root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# java -version
openjdk version "11.0.22" 2024-01-16 LTS
OpenJDK Runtime Environment Zulu11.70+15-CA (build 11.0.22+7-LTS)
OpenJDK 64-Bit Server VM Zulu11.70+15-CA (build 11.0.22+7-LTS, mixed mode)
```

### CODE REVIEW

**Comment by aooohan** [comment]
Great!   Although I haven't reviewed it yet, from the log you posted, I found a problem. ;)
> ```
> root@yimiao-ltp:/home/yimiao/workspace/version-fox-plugins/java# vfox use java@11
> Warning: The current shell lacks hook support or configuration. It has switched to global scope automatically.
> ```

If your shell prompt `Warning:The current shell lacks hook support or configuration. It has switched to global scope automatically` that means you do not hook `vfox` into your shell, please hook it manually first.

It works, but it's not a good experience.

File: java/azul-jdk.lua [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+--
+--   Unless required by applicable law or agreed to in writing, software
+--   distributed under the License is distributed on an "AS IS" BASIS,
+--   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+--   See the License for the specific language governing permissions and
+--   limitations under the License.
+
+local http = require("http")
+local json = require("json")
+local html = require("html")
```
Comment (aooohan) [inline java/azul-jdk.lua]: This html module is not used, it needs to be removed.

Comment (aooohan) [inline java/azul-jdk.lua]:
```suggestion
```

**Review by aooohan — CHANGES_REQUESTED** [review]

**Comment by aooohan** [comment]
Thanks for the PR. ;)

File: java/azul-jdk.lua (hunk 1 above)
Comment (yimiaoxiehou) [reply java/azul-jdk.lua]: 春节快乐，放假了没看邮件  ^-^

**Pull request merged**